src/reports/library/reports.cpp

```diff
@@ -0,0 +1,176 @@
+#include <bbp/sonata/reports.h>
+#include <iostream>
+#include <reports/library/sonatareport.hpp>
+#include <reports/utils/logger.hpp>
+
+using namespace bbp::sonata;
+
+SonataReport sonata_report;
+
+int sonata_clear() {
+    sonata_report.clear();
+    return 0;
+}
+
+int sonata_create_report(
+    const char* report_name, double tstart, double tend, double dt, const char* kind) {
+    try {
+        if (!sonata_report.report_exists(report_name)) {
+            sonata_report.create_report(report_name, kind, tstart, tend, dt);
+        }
+    } catch (const std::exception& err) {
+        logger->error(err.what());
+    }
+    return 0;
+}
+
+int sonata_add_node(const char* report_name, uint64_t node_id) {
+    try {
+        if (sonata_report.report_exists(report_name)) {
+            auto report = sonata_report.get_report(report_name);
+            report->add_node(node_id);
+        }
+    } catch (const std::exception& err) {
+        logger->error(err.what());
+    }
+    return 0;
+}
+
+int sonata_add_element(const char* report_name,
+                       uint64_t node_id,
+                       uint32_t element_id,
+                       double* voltage) {
+    if (!sonata_report.report_exists(report_name)) {
+        return 0;
```

Comment:
Should be also non 0 I suppose, please check the other function return values in this module.